packages/odie-client/src/components/message/sources.tsx

```diff
@@ -14,13 +14,21 @@ export const Sources = ( { message }: { message: Message } ) => {
 		if ( messageLength > 0 ) {
 			// Record TrainTracks render events
 			message.context?.sources?.forEach( ( source: Source, index: number ) => {
-				trackEvent( 'sources_traintracks_render', {
-					fetch_algo: source?.railcar?.fetch_algo,
-					ui_algo: 'default',
-					railcar: source?.railcar?.railcar,
-					fetch_position: source?.railcar?.fetch_position,
-					ui_position: index,
-				} );
+				if ( source.railcar ) {
+					trackEvent( 'sources_traintracks_render', {
+						fetch_algo: source?.railcar?.fetch_algo,
+						ui_algo: 'default',
+						message_id: message?.message_id,
+						railcar: source?.railcar?.railcar,
+						fetch_position: source?.railcar?.fetch_position,
+						fetch_query: source?.railcar?.fetch_query,
+						fetch_lang: source?.railcar?.fetch_lang,
+						ui_position: index,
+						rec_query: source?.railcar?.rec_query,
```

Comment:
We are missing tis property in the typing

Reply:
Actually, after reviewing again, I realized that the backend never returns `rec_query`, only `fetch_query`, so I just removed it.